src/components/Quiz/useLocalQuizData.ts

```diff
@@ -1,28 +1,25 @@
-import { CompletedQuizzes, UserStats } from "@/lib/types"
-
-import allQuizzesData from "@/data/quizzes"
+import { UserStats } from "@/lib/types"
 
 import { USER_STATS_KEY } from "@/lib/constants"
 
 import { useLocalStorage } from "@/hooks/useLocalStorage"
 
-/**
- * Contains each quiz id as key and <boolean, number> to indicate if its completed and the highest score in that quiz
- *
- * Initialize all quizzes as not completed
- */
-const INITIAL_COMPLETED_QUIZZES: CompletedQuizzes = Object.keys(
-  allQuizzesData
-).reduce((object, key) => ({ ...object, [key]: [false, 0] }), {})
-
 export const INITIAL_USER_STATS: UserStats = {
   score: 0,
   average: [],
-  completed: INITIAL_COMPLETED_QUIZZES,
+  completed: {},
 }
 
 export const useLocalQuizData = () => {
   const data = useLocalStorage(USER_STATS_KEY, INITIAL_USER_STATS)
 
+  // If the user has an old version of the app, convert the
+  // `completed` value from a string to an object.
+  const [current, setCurrent] = data
+  if (typeof current.completed === "string") {
+    const newCompleted = JSON.parse(current.completed)
+    setCurrent({ ...current, completed: newCompleted })
+  }
```

Comment:
@pettinarip @TylerAPfledderer This should fix the issues with the escaped strings for the `completed` value. This was happening because the `completed` object was being stringified inside a parent also being stringified.


This simply checks if `.completed` is of `string` type, and if it is, parses it and re-assigns as an `object` to the locale storage value.


```
// If "current" value looks like this:
{"score":3,"average":[60],"completed":"{\"what-is-ethereum\":[false,0],\"what-is-ether\":[false,0],\"web3\":[false,0],\"wallets\":[false,0],\"security\":[false,0],\"nfts\":[false,3],\"layer-2\":[false,0],\"merge\":[false,0]}"}

// ...this code block will convert it to this:
{"score":3,"average":[60],"completed":{"what-is-ethereum":[false,0],"what-is-ether":[false,0],"web3":[false,0],"wallets":[false,0],"security":[false,0],"nfts":[false,3],"layer-2":[false,0],"merge":[false,0]}}
```

Just need to double check no edge cases if we have this local storage data and then take a stand-alone quiz, but hopefully this fixes the backward compatibility.

Reply:
Great! yeah, I guess we will be forced to do this for now.

Reply:
This might also help with the setup with the `QuizManager` waiting to come online (To have the quiz components isolated and to be supplied separate groups of quiz data) as well as that issue where adding quizzes or quiz questions to the data set compromised existing user stats and throwing id errors.

Does this help with the later?

Reply:
This fix should take care of compatibility issues when adding new quizzes to the data set.